### PR TITLE
Make rich links accessible

### PR DESCRIPTION
**Pull request by MarSavar**
## What does this change?
Makes rich links accessible by:

- Using a light grey background in all cases (previously, Editorial/Comment/Letter designs used a darker background)
- Making the "Read more" text bold
- Making the text and the SVG visible in dark mode before the client-side JS script that applies the class name kicks-in (see GIF). Previously, it was black on dark grey which was barely visible. Not a huge difference since the JS code runs almost instantaneously once the page loads, but it was slightly annoying.

## Why?
We were failing various accessibility checks, particularly in the comment design with orange text on dark grey background. I also investigated other themes and even though the contrast was ok, the fact that the font wasn't bold was causing the check to fail. I used https://webaim.org/resources/contrastchecker/ for accessibility checks.

**Note**: This PR looks harder than it is to review mainly because I got rid of `getThemeStyles` and replaced all occurrences with `editorialPalette`, which is the way forward.

## Screenshots
| Before      | After      |
|-------------|------------|
| ![before][] | ![after][] |
| ![before2][] | ![after2][] |

[before]: https://user-images.githubusercontent.com/57295823/169103037-cec435fd-0391-4833-92a0-30307bda9180.png
[after]: https://user-images.githubusercontent.com/57295823/169103077-876e3b7b-a6a8-4c95-9a54-1de2a10c28ec.png
[before2]: https://user-images.githubusercontent.com/57295823/170326010-322eab6f-1e9f-4b9f-84cc-cb1329395d74.gif
[after2]: https://user-images.githubusercontent.com/57295823/170326285-fd73cae0-2692-48d3-80a1-884735dc1c30.gif

### CODE REVIEW

**Comment by mxdvl** [comment]
This looks nice! I wanted to point out that orange contrast is a known limitation of WCAG2, and it would be interesting to see how this contrast fares in the upcoming APCA in WCAG3:

https://gist.github.com/Myndex/069a4079b0de2930e72d5401bde9af98

File: apps-rendering/src/components/RichLink/index.tsx [hunk 1]
```diff
@@ -54,45 +58,30 @@ const liveBlogRichLinkStyles = css`
...
-	);
-	const { kicker: lifestyleKicker, inverted: lifestyleInverted } =
-		getThemeStyles(ArticlePillar.Lifestyle);
-
 	return css`
-		background: ${backgroundColor(format)};
+		background: ${background.richLink(format)};
 		padding: ${remSpace[3]} ${remSpace[3]} ${remSpace[2]};
-		border-top: solid 1px ${neutral[60]};
+		border-top: solid 1px ${border.richLinkDark(format)};
```
Comment (JamieB-gu) [inline apps-rendering/src/components/RichLink/index.tsx:64]: Dark mode colour in light mode?

File: apps-rendering/src/components/RichLink/index.tsx [hunk 2]
```diff
@@ -54,45 +58,30 @@ const liveBlogRichLinkStyles = css`
...
 
 		&.js-culture {
-			${richLinkPillarStyles(cultureKicker, cultureInverted)}
+			${richLinkPillarStyles(format)}
 		}
 
 		&.js-lifestyle {
-			${richLinkPillarStyles(lifestyleKicker, lifestyleInverted)}
+			${richLinkPillarStyles(format)}
 		}
```
Comment (JamieB-gu) [inline apps-rendering/src/components/RichLink/index.tsx:85]: Is `format` the `Format` of the main article? If so, how will rich links pick up the format of the articles they're pointing to?

File: common-rendering/src/editorialPalette/border.ts [hunk 3]
```diff
@@ -86,6 +86,58 @@ const standfirstLink = (format: ArticleFormat): Colour => {
...
+			return specialReport[500];
+	}
+};
+
+/**
+ * Neutral colour that is applied server-side before client-side JS changes the colour based on the theme.
+ */
+const richLinkSvgPreload = (format: ArticleFormat): Colour => {
+	return neutral[7];
+}
```
Comment (JamieB-gu) [inline common-rendering/src/editorialPalette/border.ts:135]: Is there a dark mode version of this?

File: common-rendering/src/editorialPalette/text.ts [hunk 4]
```diff
@@ -597,6 +597,40 @@ const seriesTitle = (format: ArticleFormat): Colour => {
...
+}
+
+const richLinkAnchorDark = (_format: ArticleFormat): Colour => {
+	return neutral[60];
+}
+
+/**
+ * Neutral colour that is applied server-side before client-side JS changes the colour based on the theme.
+ */
+ const richLinkSvgPreload = (format: ArticleFormat): Colour => {
```
Comment (JamieB-gu) [inline common-rendering/src/editorialPalette/text.ts:630]: Is there a dark mode version of this?

File: common-rendering/src/editorialPalette/text.ts [hunk 5]
```diff
@@ -597,6 +597,40 @@ const seriesTitle = (format: ArticleFormat): Colour => {
...
+		case ArticleSpecial.SpecialReport:
+			return specialReport[500];
+	}
+};
+
+const richLinkDark = (_format: ArticleFormat): Colour => {
+	return neutral[86];
+}
+
+const richLinkAnchorDark = (_format: ArticleFormat): Colour => {
```
Comment (JamieB-gu) [inline common-rendering/src/editorialPalette/text.ts:623]: Is there a light mode version of this?

**Review by JamieB-gu — COMMENTED** [review]
Looks great, 👍 for all the palette work!

Some questions/comments.

File: apps-rendering/src/components/RichLink/index.tsx (hunk 2 above)
Comment (MarSavar) [reply apps-rendering/src/components/RichLink/index.tsx:85]: Ahh good shout! 🤦 
Fixed this.

**Review by JamieB-gu — APPROVED** [review]

**Review by iainjchambers-guardian — APPROVED** [review]

**Pull request merged**